src/playground/sample_data_ingestion/sample_data_ingestor/sample_data_ingestor.py

```diff
@@ -0,0 +1,345 @@
+from datetime import datetime, timedelta
+from shutil import copyfileobj
+from json import loads, dumps
+from os import remove, path
+from random import randint
+import gzip
+import sys
+import csv
+
+# Adds parent directory "/sample_data_ingestion" to sys.path
+sys.path.append(path.dirname(path.dirname(path.abspath(__file__))))
+from sample_data_generator.sample_data_generator import generate_data
+from sample_data_generator.sample_data_generator import choose_field
+
+
+def generate_noise(initial_value, anomaly_percentage, avg_min, avg_max, abs_min, abs_max):
+    """
+    Function that generates random noise in the data trend (used for Anomaly Detection)
+
+    Arguments:
+        - initial_value: The original non-anomaly value generated from generate_data_trend()
+        - anomaly_percentage: How often an anomaly occurs (minimum: 0.001)
+        - avg_min: The lowest average value in which there is no anomaly
+        - avg_max: The highest average value in which there is no anomaly
+        - abs_min: The lowest possible value, which can be an anomaly
+        - abs_max: The highest possible value, which can be an anomaly
+    
+    Returns:
+        - A numeric value representing the new field value
+    """
+    new_value = initial_value
+    noise = randint(1, 1000) / 1000
+
+    # If an anomaly chance is enabled, change the value and return it
+    if noise <= anomaly_percentage:
```

Comment:
what's happening if nosie > anomaly_percentage?

Reply:
I left a comment in `sample_data_plugins/ad_plugin_data_config/data_trend_functions.py` but essentially, if `noise > anomaly_percentage`, the `initial_value` will not change and no anomaly will be generated for this value.